File: capitulo02_estadistica_basica/scripts/01_distribucion_normal.py
# ============================================
# Modulo 1: Validacion de Distribucion Normal
# ============================================

import numpy as np
import pandas as pd
from scipy import stats
import matplotlib.pyplot as plt
import seaborn as sns

# Configuracion
np.random.seed(123)
n = 1000

# Generar datos N(0,1)
x = np.random.normal(loc=0, scale=1, size=n)

# Estadisticas descriptivas
estadisticas = pd.DataFrame({
    'Estadistico': ['Media', 'Desv.Est.', 'Asimetria', 'Curtosis'],
    'Valor': [
        np.mean(x),
        np.std(x, ddof=1),
        stats.skew(x),
        stats.kurtosis(x, fisher=False)
    ],
    'Teorico': [0, 1, 0, 3]
})

print(estadisticas)

# Prueba de normalidad Shapiro-Wilk
w_stat, p_value = stats.shapiro(x)
print(f"\nShapiro-Wilk Test:")
print(f"W = {w_stat:.4f}")
print(f"p-value = {p_value:.4f}")

# Probabilidades y cuantiles
prob_menor_196 = stats.norm.cdf(1.96)
prob_entre_196 = stats.norm.cdf(1.96) - stats.norm.cdf(-1.96)
q_025 = stats.norm.ppf(0.025)
q_975 = stats.norm.ppf(0.975)

print(f"\nProbabilidades y Cuantiles:")
print(f"P(Z <= 1.96) = {prob_menor_196:.4f}")
print(f"P(-1.96 <= Z <= 1.96) = {prob_entre_196:.4f}")
print(f"Cuantil 2.5% = {q_025:.4f}")
print(f"Cuantil 97.5% = {q_975:.4f}")

# Visualizacion
fig, axes = plt.subplots(1, 2, figsize=(14, 6))

# Figura 1: Histograma vs densidad teorica
axes[0].hist(x, bins=30, density=True, alpha=0.7, 
             color='lightblue', edgecolor='black')
x_range = np.linspace(-4, 4, 100)
axes[0].plot(x_range, stats.norm.pdf(x_range, 0, 1), 
             'r-', linewidth=2, label='Densidad N(0,1)')
axes[0].set_xlabel('Valor')
axes[0].set_ylabel('Densidad')
axes[0].set_title('Histograma vs Densidad Teorica N(0,1)')
axes[0].legend()
axes[0].grid(alpha=0.3)

# Figura 2: Q-Q plot
stats.probplot(x, dist="norm", plot=axes[1])
axes[1].set_title('Grafico Q-Q Normal')
axes[1].grid(alpha=0.3)

plt.tight_layout()
plt.show()

File: capitulo02_estadistica_basica/scripts/01_distribucion_normal.R
# ============================================
# Modulo 1: Validacion de Distribucion Normal
# ============================================

# Configuracion
set.seed(123)
n <- 1000

# Generar datos N(0,1)
x <- rnorm(n, mean = 0, sd = 1)

# Estadisticas descriptivas
estadisticas <- data.frame(
  Estadistico = c("Media", "Desv.Est.", "Asimetria", "Curtosis"),
  Valor = c(
    mean(x),
    sd(x),
    moments::skewness(x),
    moments::kurtosis(x)
  ),
  Teorico = c(0, 1, 0, 3)
)

print(estadisticas)

# Prueba de normalidad Shapiro-Wilk
shapiro_test <- shapiro.test(x)
cat("\nShapiro-Wilk Test:\n")
cat("W =", shapiro_test$statistic, "\n")
cat("p-value =", shapiro_test$p.value, "\n")

# Probabilidades y cuantiles
prob_menor_196 <- pnorm(1.96)
prob_entre_196 <- pnorm(1.96) - pnorm(-1.96)
q_025 <- qnorm(0.025)
q_975 <- qnorm(0.975)

cat("\nProbabilidades y Cuantiles:\n")
cat("P(Z <= 1.96) =", prob_menor_196, "\n")
cat("P(-1.96 <= Z <= 1.96) =", prob_entre_196, "\n")
cat("Cuantil 2.5% =", q_025, "\n")
cat("Cuantil 97.5% =", q_975, "\n")

# Visualizacion en RStudio
library(ggplot2)

# Figura 1: Histograma vs densidad teorica
p1 <- ggplot(data.frame(x = x), aes(x = x)) +
  geom_histogram(aes(y = after_stat(density)), 
                 bins = 30, fill = "orange", 
                 color = "black", alpha = 0.7) +
  stat_function(fun = dnorm, args = list(mean = 0, sd = 1),
                color = "red", size = 1.2) +
  labs(title = "Histograma vs Densidad Teorica N(0,1)",
       x = "Valor", y = "Densidad") +
  theme_minimal()

# Mostrar primera figura en panel de plots
print(p1)

# Pausa para ver la primera gráfica (opcional)
cat("\nPresiona Enter para ver la siguiente gráfica...")
invisible(readline())

# Figura 2: Q-Q plot
p2 <- ggplot(data.frame(x = x), aes(sample = x)) +
  stat_qq() +
  stat_qq_line(color = "red", size = 1) +
  labs(title = "Grafico Q-Q Normal",
       x = "Cuantiles Teoricos", y = "Cuantiles Muestrales") +
  theme_minimal()

# Mostrar segunda figura en panel de plots
print(p2)

# Opción alternativa: Mostrar ambas en una sola ventana
# Si prefieres ver las dos gráficas juntas, puedes usar grid.arrange
cat("\n\n¿Quieres ver ambas gráficas juntas? (s/n): ")
respuesta <- readline()

if (tolower(respuesta) == "s") {
  library(gridExtra)
  p_combinado <- grid.arrange(p1, p2, ncol = 2)
  print(p_combinado)
}
